### PR TITLE
59 admin changes order status

### PR DESCRIPTION
**Pull request by noobjey**
- user can see the date an order status is updated
- refactored index action in order controller

### CODE REVIEW

**Comment by noobjey** [comment]
I just made sure the updated_at date was displaying correctly. the test isn't great because i couldn't figure out how to set the timezone. tried in application.rb but never got active record to work correctly, always displayed in UTC. Let me know if you know a better way to do it.

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -13,11 +13,8 @@ def create
   end
 
   def index
-    if current_user
-      @orders = current_user.orders
-    else
-      authorization_error
-    end
+    @orders = current_user.orders if current_user
```
Comment (applegrain) [inline app/controllers/orders_controller.rb]: what happens if we do --> 
@orders = current_user.orders || authorization_error? 
If the expression on the left evaluates to false, will it render the authorization error?

File: spec/features/user_views_past_orders_spec.rb [hunk 2]
```diff
@@ -3,7 +3,12 @@
 RSpec.describe "a user with past orders", type: :feature do
   let!(:item) { Fabricate(:item) }
   let!(:user) { Fabricate(:user) }
+  let!(:status) { Status.create(name: "pending") }
+  let!(:order) { Order.create(user_id: user.id, status_id: status.id) }
 
+  before(:each) do
```
Comment (applegrain) [inline spec/features/user_views_past_orders_spec.rb]: is this before block empty?

File: spec/features/user_views_past_orders_spec.rb [hunk 3]
```diff
@@ -40,15 +45,26 @@
         within(".orders") do
           expect(page).to have_content("Order Number")
           expect(page).to have_content("Total")
-          expect(page).to have_content("Order Date")
+          expect(page).to have_content("Order Updated Date")
```
Comment (applegrain) [inline spec/features/user_views_past_orders_spec.rb]: Could we change these hardcoded values and use either item, user, status or order to render the values?

File: app/controllers/orders_controller.rb (hunk 1 above)
Comment (noobjey) [inline app/controllers/orders_controller.rb]: good call. added terinary instead

File: spec/features/user_views_past_orders_spec.rb (hunk 3 above)
Comment (noobjey) [inline spec/features/user_views_past_orders_spec.rb]: did it. good call.

File: spec/features/user_views_past_orders_spec.rb (hunk 2 above)
Comment (noobjey) [inline spec/features/user_views_past_orders_spec.rb]: tired commit, fixed

File: spec/features/user_views_past_orders_spec.rb [hunk 4]
```diff
@@ -40,15 +42,26 @@
         within(".orders") do
           expect(page).to have_content("Order Number")
           expect(page).to have_content("Total")
-          expect(page).to have_content("Order Date")
-          expect(page).to have_link("1")
-          expect(page).to have_link("2")
-          expect(page).to have_content("36")
-          expect(page).to have_content("12")
+          expect(page).to have_content("Order Updated Date")
```
Comment (applegrain) [inline spec/features/user_views_past_orders_spec.rb]: sweet!

File: app/controllers/orders_controller.rb [hunk 5]
```diff
@@ -13,11 +13,7 @@ def create
   end
 
   def index
-    if current_user
-      @orders = current_user.orders
-    else
-      authorization_error
-    end
+    current_user ? @orders = current_user.orders : authorization_error
```
Comment (applegrain) [inline app/controllers/orders_controller.rb]: Looks good. People don't like ternaries but I think this one is simple enough to pass.

**Comment by applegrain** [comment]
Good job. I'll merge this in.

**Pull request merged**